tests/util/event_loop.cpp

```diff
@@ -39,6 +40,29 @@
 
 using namespace realm::util;
 
+template <typename Desired, typename Actual>
+void static_assert_EventLoopDispatcher_guide(const EventLoopDispatcher<Actual>&) {
+    static_assert(std::is_same_v<Actual, Desired>);
+}
+
+[[maybe_ununsed]] void check_EventLoopDispatcher_guides() {
```

Comment:
Even though this is just a compile-time check I think it'd be better to put it in one of the files for tests?

Reply:
Oops. I saw it was in `tests/`, but I guess this is utilities *for* tests, not test *of* utilities...

Is there a good existing test file for this stuff or should I make a new one?

Reply:
There probably isn't a good pre-existing location. There aren't currently any tests for this, and EventLoopDispatcher currently actually isn't even being built on the v10 branch since the things in object store which were using it were deleted.

Reply:
Do you mind if I leave it here for now then, and it can be moved out once the dust settles? As of my latest patch it is now in an anonymous namespace, so it shouldn't even generate any actual code in the object files.

Reply:
Yeah, it's no big deal.